src/app/themes.tsx

```diff
@@ -3,25 +3,65 @@
 import GlobalStyle from "@/app/global_styles";
 import isPropValid from "@emotion/is-prop-valid";
 import { useServerInsertedHTML } from "next/navigation";
-import React, { useState } from "react";
+import React, { createContext, useState } from "react";
 import { ServerStyleSheet, StyleSheetManager, ThemeProvider } from "styled-components";
 
-export const currentTheme = {
-    foregroundColor: "#000",
-    backgroundColor: "#DDD",
-    fillColor: "lightgreen",
-    fonts: ["Comic Sans MS", "Comic Sans", "cursive"],
+export type CustomTheme = {
+    foregroundColor: string;
+    backgroundColor: string;
+    primaryForegroundColor: string;
+    primaryBackgroundColor: string;
+    secondaryForegroundColor: string;
+    secondaryBackgroundColor: string;
+    accentBackgroundColor: string;
+    accentForegroundColor: string;
+    surfaceForegroundColor: string;
+    surfaceBackgroundColor: string;
+    errorColor: string;
+};
+
+export const lightTheme: CustomTheme = {
+    foregroundColor: "#000000",
+    backgroundColor: "#eeeeee",
+    primaryForegroundColor: "#000000",
+    primaryBackgroundColor: "#63a036",
+    secondaryForegroundColor: "#ffffff",
+    secondaryBackgroundColor: "#066940",
+    accentBackgroundColor: "#609fd3",
+    accentForegroundColor: "#ffffff",
+    surfaceForegroundColor: "#000000",
+    surfaceBackgroundColor: "#ffffff",
+    errorColor: "#ff624e",
+};
+
+export const darkTheme: CustomTheme = {
+    foregroundColor: "#ffffff",
+    backgroundColor: "#1a1a1a",
+    primaryForegroundColor: "#ffffff",
+    primaryBackgroundColor: "#066940",
+    secondaryForegroundColor: "#000000",
+    secondaryBackgroundColor: "#63a036",
+    accentBackgroundColor: "#1b629c",
+    accentForegroundColor: "#ffffff",
+    surfaceForegroundColor: "#ffffff",
+    surfaceBackgroundColor: "#161414",
+    errorColor: "#ff624e",
 };
 
 interface Props {
     children: React.ReactElement;
+    theme?: CustomTheme;
 }
 
+export const ThemeUpdateContext = createContext((theme: CustomTheme) =>
+    console.error(`attempted to set theme outside of a ThemeUpdateContext.Provider: ${theme}`)
```

Comment:
Is it worth this throwing an error instead of just `console.error`'ing? It's very easy to just not look at `console.log` :)